.azure-pipelines/security-tools.yml

```diff
@@ -52,6 +52,25 @@ jobs:
       outputFormat: sarif
       scanFolder: SecurityTmp
       suppressionsFile: tools/SecurityTools/CredScanSuppressions.json
+  - task: NuGetAuthenticate@1
```

Comment:
why this authenticate task is needed if the nuget package is from external source?

Reply:
You're right. Both task `NuGetAuthenticate` and env `GUARDIAN_DEFAULT_PACKAGE_SOURCE_SECRET` are not necessary for external source. Remove them. The new test build is [here](https://dev.azure.com/azclitools/internal/_build/results?buildId=188425&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=b7ac53b1-fc2a-53cb-ebde-ce59c17928a5).